src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SNSService.kt

```diff
@@ -65,3 +68,32 @@ class SNSReferralService(
     }
   }
 }
+
+@Service
+class SNSAppointmentService(
+  private val snsPublisher: SNSPublisher,
+) : ApplicationListener<AppointmentEvent>, SNSService {
+
+  @AsyncEventExceptionHandling
+  override fun onApplicationEvent(event: AppointmentEvent) {
+    when (event.type) {
+      AppointmentEventType.ATTENDANCE_RECORDED -> {
+        val referral = event.appointment.actionPlan.referral
+        val eventType = "intervention.session-appointment.${when (event.appointment.attended) {
+          Attended.YES -> "attended"
+          Attended.LATE -> "late"
+          Attended.NO -> "missed"
```

Comment:
From a domain perspective, the two big events probation cares about are "attended" and "missed"
Perhaps "late" is just an `arrivedLate: true/false` on the "attended" payload?

Reply:
fine with me 👍